src/main/java/com/coveros/selenified/OutputFile.java

```diff
@@ -688,27 +688,29 @@ private void setStartTime() {
      *               hashmap
      * @return String: a 'prettily' formatted string that is HTML safe to output
      */
-    public String outputRequestProperties(Request params) {
-        if (params == null) {
-            return "";
-        }
+    public String outputRequestProperties(Request params, File file) {
         StringBuilder output = new StringBuilder();
-        output.append("<br/> with parameters: ");
-        output.append("<div><i>");
-        if (params.getData() != null) {
-            Gson gson = new GsonBuilder().setPrettyPrinting().create();
-            output.append(gson.toJson(params.getData()));
-        }
-        if (params.getParams() != null) {
-            for (Map.Entry<String, String> entry : params.getParams().entrySet()) {
-                output.append("<div>");
-                output.append(entry.getKey());
-                output.append(" : ");
-                output.append(entry.getValue());
-                output.append("</div>");
+        if (params != null && params.isPayload()) {
+            output.append("<br/> with parameters: ");
+            output.append("<div><i>");
+            if (params.getJsonPayload() != null) {
+                Gson gson = new GsonBuilder().setPrettyPrinting().create();
+                output.append(gson.toJson(params.getJsonPayload()));
             }
+            if (params.getMultipartData() != null) {
+                for (Map.Entry<String, Object> entry : params.getMultipartData().entrySet()) {
+                    output.append("<div>");
+                    output.append(String.valueOf(entry.getKey()));
+                    output.append(" : ");
+                    output.append(String.valueOf(entry.getValue()));
+                    output.append("</div>");
+                }
+            }
+            output.append("</i></div>");
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Define a constant instead of duplicating this literal "</i></div>" 3 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://localhost:9000/sonar/coding_rules#rule_key=squid%3AS1192)